### PR TITLE
Add swapper

### PR DESCRIPTION
**Pull request by sullof**
1. Adding a SynSwapper contract to allow an address which has the roles 
  `ssyn.ROLE_WHITE_LISTED_RECEIVER`
and
  `syn.ROLE_TREASURY`
to swap sSYN and receive SYN.
The swapper contract must be a 
  `ssyn.ROLE_TOKEN_DESTROYER`
and 
  `syn.ROLE_TOKEN_CREATOR`

2. Moving the vesting contracts in the vesting folder

### CODE REVIEW

File: contracts/token/SynSwapper.sol [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    ssyn = _ssyn;
+  }
+
+  /**
+   * @notice Swaps an amount of sSYN for an identical amount of SYN
+   * @param recipient  The address of account to burn the sSYN and receive the SSYN
+   * @param amount     The amount of token to be swapped
+   */
+  function swap(address recipient, uint256 amount) external {
+    require(isSenderInRole(ROLE_ACCESS_MANAGER), "sSYN: insufficient privileges (ROLE_ACCESS_MANAGER required)");
```
Comment (superpower007) [inline contracts/token/SynSwapper.sol:41]: why ROLE_ACCESS_MANAGER

File: contracts/token/SyndicateERC20.sol [hunk 2]
```diff
@@ -295,7 +295,13 @@ contract SyndicateERC20 is AccessControl {
    */
   uint32 public constant ROLE_ERC20_SENDER = 0x0008_0000;
 
-  uint32 public constant ROLE_WHITE_LISTED_SPENDER = 0x0010_0000;
+  uint32 public constant ROLE_WHITE_LISTED_SPENDER = 0x0016_0000;
```
Comment (superpower007) [inline contracts/token/SyndicateERC20.sol:298]: this is hex, use 0x0020_0000 maybe be more fitting?

File: contracts/token/SyndicateERC20.sol [hunk 3]
```diff
@@ -295,7 +295,13 @@ contract SyndicateERC20 is AccessControl {
...
+
+  /**
+ * @notice White listed treasury can swap sSYN to receive SYN
+ * @dev Role ROLE_TREASURY can get SYN in exchange for sSYN
+ */
+  uint32 public constant ROLE_TREASURY = 0x0032_0000;
```
Comment (superpower007) [inline contracts/token/SyndicateERC20.sol:304]: 0x0040_000 ?

File: contracts/token/SynSwapper.sol (hunk 1 above)
Comment (sullof) [reply contracts/token/SynSwapper.sol:41]: Only the superadmin can make the swap.
Alternatively, we can make that treasury can do it.

File: contracts/token/SyndicateERC20.sol (hunk 2 above)
Comment (sullof) [reply contracts/token/SyndicateERC20.sol:298]: I see but I do not understand why before they go 0002 0004 0008 and later 0010 0020. It does not make much sense for me. The problem is when you make the sum of it. I sense that there is an error there.

File: contracts/token/SyndicateERC20.sol (hunk 3 above)
Comment (sullof) [reply contracts/token/SyndicateERC20.sol:304]: Same as above

**Pull request merged**